src/store.c

```diff
@@ -237,7 +237,7 @@ static CK_RV p11prov_store_ctx_add_key(void *pctx, CK_OBJECT_CLASS class,
 
     if ((ctx->num_objs % OBJS_ALLOC_SIZE) == 0) {
         P11PROV_OBJ **tmp =
-            OPENSSL_realloc(ctx->objects, ctx->num_objs + OBJS_ALLOC_SIZE);
+            OPENSSL_realloc(ctx->objects, sizeof(P11PROV_OBJ*[ctx->num_objs + OBJS_ALLOC_SIZE]));
```

Comment:
I do not fully understand this, we are just creating an array of pointers here.
I never seen this casting within a sizeof, it confuses me.

Sounds to me this would be more understandable:
 (ctx->num_objs + OBJS_ALLOC_SIZE) * sizeof(P11PROV_OBJ *)
Ie:
OPENSSL_realloc(ctx->objects, (ctx->num_objs + OBJS_ALLOC_SIZE) * sizeof(P11PROV_OBJ *));

Reply:
`sizeof` takes size of the type. And the cast looking thing is type definition of the array we are instantiating. For me this is more readable and safe (no parenthesis required, taking size of type instead of calculating it), than doing the multiplication manually, but I'll change it.